### PR TITLE
file:truncate added to file module, with requested changes

### PR DESCRIPTION
**Pull request by supreetsingh10**
Made the requested changes, also added documentation, all feedback is welcome, thank you once again.

### CODE REVIEW

**Comment by supreetsingh10** [comment]
Mr @xiaq why is this one test case failing?

File: pkg/eval/mods/file/file.go [hunk 1]
```diff
@@ -124,3 +127,55 @@ func prclose(p vals.Pipe) error {
...
+// ```elvish
+// file:truncate $filename $size
+// ```
+// Truncate changes the size of
+// the named file. If the file is
+// a symbolic link, it changes the
+// size of the link's target.
+//
+// If there is an error it will
+// be of type *PathError.
```
Comment (xiaq) [inline pkg/eval/mods/file/file.go:142]: This seems to be copied verbatim from https://pkg.go.dev/os#Truncate; Go docs have different conventions to elvdocs and I'll amend this.

File: pkg/eval/mods/file/file.go [hunk 2]
```diff
@@ -124,3 +127,55 @@ func prclose(p vals.Pipe) error {
...
+	var offset int64
+	switch size.(type) {
+	case int:
+		offset = int64(size.(int))
+		if offset < 0 {
+			return errs.BadValue{
+				What:   "Numeric input for the function is wrong",
+				Valid:  "positive 64 bit integer",
+				Actual: "Number less than 0",
+			}
```
Comment (xiaq) [inline pkg/eval/mods/file/file.go:154]: This should use errs.OutOfRange. The multiple checks against negative values should also be merged together.

File: pkg/eval/mods/file/file.go [hunk 3]
```diff
@@ -124,3 +127,55 @@ func prclose(p vals.Pipe) error {
...
+// the named file. If the file is
+// a symbolic link, it changes the
+// size of the link's target.
+//
+// If there is an error it will
+// be of type *PathError.
+
+func truncate(name string, size vals.Num) error {
+	var offset int64
+	switch size.(type) {
```
Comment (xiaq) [inline pkg/eval/mods/file/file.go:146]: Using an assigned type switch helps avoid type assertions below:

```go
switch size := size.(type)
```

**Review by xiaq — APPROVED** [review]
The CI is failing because the overall test coverage will decrease - there is no test for the newly introduced function.

But I'll merge this anyway and do a fixup commit.

**Pull request merged**